jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -174,7 +175,8 @@ void reportSpan(Span span) {
     public io.opentracing.Tracer.SpanBuilder addReference(
         String referenceType, io.opentracing.SpanContext referencedContext) {
       if (parent == null
-          && (referenceType == References.CHILD_OF || referenceType == References.FOLLOWS_FROM)) {
+          && (Objects.equals(referenceType, References.CHILD_OF)
+              || Objects.equals(referenceType, References.FOLLOWS_FROM))) {
```

Comment:
the use of `==` was definitely a bug, would only work for enums, but the constants are Strings.

But it doesn't mean we need to use `Objects.equals()`, instead use `References.CHILD_OF.equals(...)`